### PR TITLE
nrf_rpc: add documentation page for the UART transport

### PR DESCRIPTION
**Pull request by Damian-Nordic**
Add a documentation page that describes the nRF RPC UART transport frame encoding and API.

### CODE REVIEW

File: doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+* If the byte matches one of the :ref:`special octets <nrf_rpc_uart_special_octets>`, it shall be encoded as the following two octets:
+
+  * the **escape** octet (``0x7d``),
+  * the input byte XORed with ``0x20``.
+
+* Otherwise, the byte shall be encoded into a frame’s octet without changes.
+
+.. note::
+
+   The current frame format is transitional and will be extended with CRC and other fields to ensure the transport reliability.
```
Comment (wiba-nordic) [inline doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst:31]:
```suggestion
   The current frame format is transitional and will be extended with CRC and other fields to ensure transport reliability.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]
Is this not aimed at the `main` branch on purpose?

**Comment by Damian-Nordic** [comment]
> Is this not aimed at the `main` branch on purpose?

Yes, we're developing a proof of concept on branch but it will be merged to main at some point.

**Review by wiba-nordic — APPROVED** [review]

**Comment by canisLupus1313** [comment]
@Damian-Nordic Overall looks good, but would be nice to add that this version is experimental, and will change.

**Comment by Damian-Nordic** [comment]
> @Damian-Nordic Overall looks good, but would be nice to add that this version is experimental, and will change.

You mean apart from this big note I added:
"The current frame format is transitional and will be extended with CRC and other fields to ensure transport reliability."? ;)

**Review by bpienk — APPROVED** [review]

**Comment by canisLupus1313** [comment]
> > @Damian-Nordic Overall looks good, but would be nice to add that this version is experimental, and will change.
> 
> You mean apart from this big note I added: "The current frame format is transitional and will be extended with CRC and other fields to ensure transport reliability."? ;)
> 
> Or maybe should I move it to the section beginning?

I think it would be more visible on top of the sesction.

**Review by canisLupus1313 — APPROVED** [review]

**Comment by wiba-nordic** [comment]
> > > @Damian-Nordic Overall looks good, but would be nice to add that this version is experimental, and will change.
> > 
> > 
> > You mean apart from this big note I added: "The current frame format is transitional and will be extended with CRC and other fields to ensure transport reliability."? ;)
> > Or maybe should I move it to the section beginning?
> 
> I think it would be more visible on top of the sesction.

I'd say adding it after the first line in the section would be best for the meaning, since that puts it just before the description of the current implementation.

**Pull request merged**